File: app/src/main/java/com/eddief/android/wallofaww/model/AwwRoomDB.java
package com.eddief.android.wallofaww.model;

import android.arch.persistence.room.Database;
import android.arch.persistence.room.Room;
import android.arch.persistence.room.RoomDatabase;
import android.content.Context;

@Database(entities = {AwwImage.class}, version = 1, exportSchema = false)
public abstract class AwwRoomDB extends RoomDatabase{

    public abstract ImageDao imageDao();
    private static AwwRoomDB INSTANCE;

    public static AwwRoomDB getDatabase(final Context context) {
        if (INSTANCE == null) {
            synchronized (AwwRoomDB.class) {
                if (INSTANCE == null) {
                    INSTANCE = Room.databaseBuilder(context.getApplicationContext(),
                            AwwRoomDB.class, "aww_database")
                            .build();
                }
            }
        }
        return INSTANCE;
    }

}

File: app/src/main/java/com/eddief/android/infiniteaww/data/AwwRoomDB.kt
package com.eddief.android.infiniteaww.data

import android.content.Context
import androidx.room.Database
import androidx.room.Room
import androidx.room.RoomDatabase
import com.eddief.android.infiniteaww.model.AwwImage

@Database(entities = [AwwImage::class], version = 1, exportSchema = false)
abstract class AwwRoomDB : RoomDatabase() {
    abstract fun imageDao(): ImageDao

    companion object {
        private var INSTANCE: AwwRoomDB? = null
        fun getDatabase(context: Context): AwwRoomDB {
            if (INSTANCE == null) {
                synchronized(AwwRoomDB::class.java) {
                    if (INSTANCE == null) {
                        INSTANCE = Room.databaseBuilder(
                            context.applicationContext,
                            AwwRoomDB::class.java, "aww_database"
                        ).build()
                    }
                }
            }
            return INSTANCE!!
        }
    }
}